python/dgl/graph.py

```diff
@@ -1165,6 +1170,10 @@ def from_networkx(self, nx_graph, node_attrs=None, edge_attrs=None):
                 [2., 2., 2., 2.],
                 [1., 1., 1., 1.]])
         """
+        nx_graph = nx_graph.to_directed()
```

Comment:
What will happen for the node/edge attributes after `to_directed()`?

Reply:
`to_directed` returns a deep copy of the original undirected graph where all `(node, node_attrs)` and `(edge, edge_attrs)` pairs are preserved. Besides, originally the undirected graph only has `u, v` and its associated attributes. `to_directed` creates `v, u` edges whose attributes are the same as `u, v`. See [`to_directed` source code](https://github.com/networkx/networkx/blob/master/networkx/classes/graph.py#L1602). This is useful as otherwise we will get `None` attributes for the new `v, u` edges and an error will occur in [L1178 of graph.py](https://github.com/dmlc/dgl/blob/master/python/dgl/graph.py#L1178) as you are concatenating tensors with None type objects.

Reply:
I will add a condition check as the function seems still returns a deepcopy even if the graph is already directed.

Reply:
Got that. Maybe add some comments about this behavior?

Reply:
Comments added. I also added the condition of checking whether a networkx graph is directed in `graph_index.py` and `immutable_graph_index.py`.